src/Query/Operation/AbstractOperation.php

```diff
@@ -0,0 +1,41 @@
+<?php
+declare(strict_types=1);
+
+namespace TBolier\RethinkQL\Query\Operation;
+
+use TBolier\RethinkQL\Query\AbstractQuery;
+
+abstract class AbstractOperation extends AbstractQuery
+{
+    /**
+     * @inheritdoc
+     */
+    public function count(): AbstractOperation
```

Comment:
Wouldn't it be nicer if these would have an Interface as return type? Nitpicking maybe ;)

Reply:
Done and correctly set the return types based on the right interface.